### PR TITLE
docs: add Depends on section, mention some inter-check dependencies

### PR DESCRIPTION
**Pull request by xginn8**
Connects-to: #172
Change-type: patch
Signed-off-by: Matthew McGinn <matthew@balena.io>

### CODE REVIEW

**Comment by xginn8** [comment]
@garethtdavies specifically, I'd love to hear your thoughts on the empty/blank `Depends on` stanzas vs omitting them entirely...

File: diagnostics.md [hunk 1]
```diff
@@ -10,31 +10,38 @@ everyone, while typically command output is used by support/subject matter exper
...
+been removed from production.
 
 As of May 1, 2019, `balenaOS 1.x` has been deprecated. These OSes are now unsupported. For more information, read our
 blog post: https://www.balena.io/blog/all-good-things-come-to-an-end-including-balenaos-1-x/.
 
 #### Triage
 Upgrade your device to the latest `balenaOS 2.x` (contact support if running 1.x).
 
+#### Depends on
+Parts of this check depend on fully functional networking stack (see [check_networking](#check_networking)).
```
Comment (garethtdavies) [inline diagnostics.md]:
```suggestion
Parts of this check depend on a fully functional networking stack (see [check_networking](#check_networking)).
```

File: diagnostics.md [hunk 2]
```diff
@@ -97,6 +115,9 @@ Confirm that NTP is not blocked at the network level, and that any specified ups
 absolutely necessary, it is possible to temporarily sync the clock using HTTP headers (though this change will not
 persist across reboots).
 
+#### Depends on
+This check depends on fully functional networking stack (see [check_networking](#check_networking)).
```
Comment (garethtdavies) [inline diagnostics.md]:
```suggestion
This check depends on a fully functional networking stack (see [check_networking](#check_networking)).
```

File: diagnostics.md (hunk 1 above)
Comment (xginn8) [reply diagnostics.md]: derp, nice catch

**Pull request merged**